ci/unit_tests/compare_dialogs/main_test.py

```diff
@@ -0,0 +1,150 @@
+    nullJsonPath = os.path.abspath(os.path.join(dataBasePath, 'null.json'))
+    valueJsonPath = os.path.abspath(os.path.join(dataBasePath, 'value.json'))
+
+    def callfunc(self, *args, **kwargs):
+        compare_dialogs.main(*args, **kwargs)
+
+    ''' Tests some basic sets of args '''
+    def test_args(self):
+        self.tTooFewArgs([])
+        self.tTooFewArgs(['/some/random/path'])
+        self.tUnrecognizedArgs(['/some/random/path', '-s', 'randomNonPositionalArg'])
+
+    ''' Tests if the first file does not exist '''
+    def test_nonExistentFileFirst(self):
+        self.tExitCodeAndErrMessage(
+            1, # exit code
+            'Input dialog json \'/some/random/path\' does not exist.', # error message substring
+            [
+                '/some/random/path',
+                ''
+            ] # params (*args, **kwargs)
+        )
+
+    ''' Tests if the second file does not exist '''
+    def test_nonExistentFileSecond(self):
+        self.tExitCodeAndErrMessage(
+            1, # exit code
+            'Output dialog json \'/some/random/path\' does not exist.', # error message substring
+            [
+                self.noJsonPath,
+                '/some/random/path'
+            ] # params (*args, **kwargs)
+        )
+
+    ''' Tests if the first file contains invalid json '''
+    def test_invalidJsonFirst(self):
+        self.tRaiseError(
+            ValueError, # exeption
+            'No JSON object could be decoded', # error message substring
+            [
+                self.noJsonPath,
+                self.emptyDictJsonPath,
+            ] # params (*args, **kwargs)
+        )
+
+    ''' Tests if the second file contains invalid json '''
+    def test_invalidJsonSecond(self):
+        self.tRaiseError(
+            ValueError, # exeption
+            'No JSON object could be decoded', # error message substring
+            [
+                self.emptyDictJsonPath,
+                self.noJsonPath,
+            ] # params (*args, **kwargs)
+        )
+
+    ''' Tests if basic jsons are same - null, null '''
+    def test_compareBasicNullNull(self):
+        self.tExitCode(0, [self.nullJsonPath, self.nullJsonPath])
+
+    ''' Tests if basic jsons are same - value, value '''
+    def test_compareBasicValueValue(self):
+        self.tExitCode(0, [self.valueJsonPath, self.valueJsonPath])
+
+    ''' Tests if basic jsons are same - empty dict, empty dict'''
+    def test_compareBasicEmptyDictEmptyDict(self):
+        self.tExitCode(0, [self.emptyDictJsonPath, self.emptyDictJsonPath])
+
+    ''' Tests if basic jsons are same - empty list, empty list'''
+    def test_compareBasicEmptyListEmptyList(self):
+        self.tExitCode(0, [self.emptyListJsonPath, self.emptyListJsonPath])
+
+    ''' Tests if basic jsons are different - null, value and value, null '''
+    def test_compareBasicNullValue(self):
+        self.tExitCode(1, [self.nullJsonPath, self.valueJsonPath])
```

Comment:
I think the return code should be different when the JSONs differ and when there is an error (invalid JSON or non-existent path). My recommendation is `1` for different JSONs and `2` for errors (I think `grep` has same convention).

Reply:
@Tereza-Pytelova @ota-sandr we should discuss this to have same behavior across all scripts

Reply:
Issue created: #245